subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -69,16 +94,285 @@ static void on_modem_lib_init(int ret, void *ctx)
 	modem_lib_init_result = ret;
 }
 
+
+static int loaded(void)
+{
+	return 0;
+}
+
+static int *target_image_type_buffer_get(uint16_t instance)
+{
+	if (instance >= FOTA_INSTANCE_COUNT) {
+		return NULL;
+	}
+
+	return &target_image_type[instance];
+}
+
+static int target_image_type_get(uint16_t instance)
+{
+	if (instance >= FOTA_INSTANCE_COUNT) {
+		return DFU_TARGET_IMAGE_TYPE_NONE;
+	}
+
+	return target_image_type[instance];
+}
+
+static int set(const char *key, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	uint16_t len;
+	void *buf;
+	struct lwm2m_obj_path path;
+	int ret;
+
+	if (!key) {
+		return -ENOENT;
+	}
+
+	LOG_DBG("Loading \"%s\"", key);
+
+	ret = lwm2m_string_to_path(key, &path, '/');
+	if (ret) {
+		return ret;
+	}
+
+	if (path.level == LWM2M_PATH_LEVEL_RESOURCE) {
+		if (lwm2m_engine_get_res_buf((char *)key, &buf, &len, NULL, NULL) != 0) {
+			LOG_ERR("Failed to get data pointer");
+			return -ENOENT;
+		}
+	} else if (path.level == LWM2M_PATH_LEVEL_OBJECT_INST) {
+		/* Image Type */
+		buf = target_image_type_buffer_get(path.obj_inst_id);
+		if (!buf) {
+			LOG_ERR("Failed to get data pointer");
+			return -ENOENT;
+		}
+		len = sizeof(int);
+	} else {
+		LOG_ERR("Unsupported path level %d", path.level);
+		return -ENOENT;
+	}
+
+	len = read_cb(cb_arg, buf, len);
+	if (len <= 0) {
+		LOG_ERR("Failed to read data");
+		return -ENOENT;
+	}
+
+	if (path.level == LWM2M_PATH_LEVEL_RESOURCE) {
+		lwm2m_engine_set_res_data_len((char *)key, len + 1);
+	}
+
+	return 0;
+}
+
+static struct settings_handler lwm2m_firm_settings = {
+	.name = LWM2M_FIRM_PREFIX,
+	.h_set = set,
+	.h_commit = loaded,
+};
+
+static const char *lwm2m_adv_path(uint16_t inst, uint16_t res)
+{
+	int ret;
+	static char path[LWM2M_MAX_PATH_STR_LEN];
+
+	ret = snprintk(path, sizeof(path), "%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT, inst, res);
+	if (ret < 0 || ret >= sizeof(path)) {
+		memset(path, 0, sizeof(path));
+	}
+	return path;
+}
+
+static int write_resource_to_settings(int inst, int res, uint8_t *data, uint16_t data_len)
+{
+	char path[sizeof(LWM2M_FIRM_PREFIX "/65632/0/10")];
```

Comment:
```suggestion
	char path[sizeof(LWM2M_FIRM_PREFIX "/65535/0/10")];
```
Code-wise it doesn't really matter but made me to stop for a second to think why the value is what it is.